packages/core/src/index.ts

```diff
@@ -904,8 +915,15 @@ class Hopr extends EventEmitter {
         log('initiating closure')
         txHash = await this.connector.initializeClosure(counterpartyPubKey)
       } else {
-        log('finalizing closure')
-        txHash = await this.connector.finalizeClosure(counterpartyPubKey)
+        // verify that we passed the closure waiting period to prevent failing
+        // on-chain transactions
+
+        if (channel.closureTimePassed()) {
+          log('finalizing closure')
+          txHash = await this.connector.finalizeClosure(counterpartyPubKey)
+        } else {
+          log('ignoring finalizing closure because closure window is still active', channel.getId())
```

Comment:
In this case we output `{ receipt: undefined; status: ChannelStatus }` instead of `{ receipt: string; status: ChannelStatus }`. Is that a problem?

Reply:
Wouldn't think so.